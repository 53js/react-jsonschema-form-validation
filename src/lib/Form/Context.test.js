import React from 'react';
import { mount } from 'enzyme';

import FormContext, { useFormContext, withFormContext } from './Context';

describe('FormContext', () => {
	it('should be a React Context', () => {
		expect(FormContext.Provider).toBeDefined();
		expect(FormContext.Consumer).toBeDefined();
	});
});

describe('useFormContext', () => {
	it('should match snapshot', () => {
		const UseContextComponent = () => {
			const context = useFormContext();
			return <span>{context}</span>;
		};
		const wrapper = mount(
			<FormContext.Provider value="FormContextSnapshot">
				<UseContextComponent />
			</FormContext.Provider>,
		);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('withFormContext', () => {
	it('should match snapshot', () => {
		const WithContextComponent = () => withFormContext((context) => <span>{context}</span>);
		const wrapper = mount(
			<FormContext.Provider value="FormContextSnapshot">
				<WithContextComponent />
			</FormContext.Provider>,
		);
		expect(wrapper).toMatchSnapshot();
	});
});
