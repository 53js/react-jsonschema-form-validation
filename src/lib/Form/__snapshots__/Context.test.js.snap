// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFormContext should match snapshot 1`] = `
<UseContextComponent>
  <span>
    FormContextSnapshot
  </span>
</UseContextComponent>
`;

exports[`withFormContext should match snapshot 1`] = `
<WithContextComponent>
  <span>
    FormContextSnapshot
  </span>
</WithContextComponent>
`;
